src/wg/util/format.d

```diff
@@ -0,0 +1,104 @@
+module wg.util.format;
+
+import wg.util.traits : isSigned;
+
+/**
+ * Format an integer as a string.
+ */
+string formatInt(I)(I i, uint minSize = 0) @safe pure nothrow
+{
+    char[21] buffer;
+    char[] r = formatInt!I(i, buffer, minSize);
+    return cast(string)null ~ r;
+}
+
+/**
+ * Format an integer as a string.
+ */
+char[] formatInt(I)(I i, ref char[21] output, uint minSize = 0) @safe pure nothrow @nogc
+{
+    assert(minSize <= 20, "Too many digits!");
+
+    uint numDigits = 0;
+    if (i == 0)
+    {
```

Comment:
I'd add `if(minSize == 0) minSize = 1;` here (or minSize set to 1 by default) else if `i` equals to zero output will be zero length.
```
unittest
{
        char[21] output;
        assert(formatInt(0, output) == "0");
}
```

Reply:
Nice catch!